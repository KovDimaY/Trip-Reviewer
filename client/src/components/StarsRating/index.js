import React, { PureComponent } from 'react';
import Rating from 'react-rating';

import './styles.css';

class StarsRating extends PureComponent {
  static getLabel(rating) {
    switch (rating) {
      case 1: return 'Hated it!';
      case 2: return 'Disliked it!';
      case 3: return 'It was ok.';
      case 4: return 'Liked it!';
      case 5: return 'Loved it!';
      default: return 'Rate your trip:';
    }
  }

  state = {
    hower: null,
  }

  emptySymbol = <i className="far fa-star star-empty" />;

  fullSymbol = <i className="fas fa-star star-full" />;

  handleRating = (rating) => {
    this.props.onChange(rating);
  }

  handleHower = (hower) => {
    this.setState({ hower });
  }

  render() {
    const { hower } = this.state;
    const { rating } = this.props;

    return (
      <div className="stars-rating-container">
        <label className="rating-label">
          {StarsRating.getLabel(hower || rating)}
        </label>
        <Rating
          initialRating={rating}
          emptySymbol={this.emptySymbol}
          fullSymbol={this.fullSymbol}
          onChange={this.handleRating}
          onHover={this.handleHower}
        />
      </div>
    );
  }
}

StarsRating.propTypes = {
  rating: React.PropTypes.number.isRequired,
  onChange: React.PropTypes.func.isRequired,
};

export default StarsRating;
