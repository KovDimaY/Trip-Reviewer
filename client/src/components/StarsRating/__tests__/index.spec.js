import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import StarsRating from './../../StarsRating';

jest.mock('./../../Sidenav/index', () => 'Nav');
jest.mock('react-router-dom', () => ({ Link: 'Link' }));

const mockComponent = props => {
    return (
        <StarsRating {...props} />
    );
};

describe('<StarsRating />', () => {
    it('should render component', () => {
        const tree = create(mockComponent()).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should change state correctly when handleHower is called', () => {
        const instance = mount(mockComponent()).instance();
        const hower = 3;

        instance.handleHower(hower);

        expect(instance.state.hower).toEqual(hower);
    });

    it('should call onChange when handleRating is called', () => {
        const onChange = jest.fn();
        const props = { onChange };
        const rating = 3;

        const instance = mount(mockComponent(props)).instance();

        instance.handleRating(rating);

        expect(onChange).toHaveBeenCalledWith(rating);
    });

    it('should return correct value when getLabel is called', () => {
        const instance = mount(mockComponent()).instance();
    
        const one = instance.getLabel(1);
        const two = instance.getLabel(2);
        const three = instance.getLabel(3);
        const four = instance.getLabel(4);
        const five = instance.getLabel(5);
        const others = instance.getLabel(100);
    
        expect(one).toEqual('Hated it!');
        expect(two).toEqual('Disliked it!');
        expect(three).toEqual('It was ok.');
        expect(four).toEqual('Liked it!');
        expect(five).toEqual('Loved it!');
        expect(others).toEqual('Rate your trip:');
    });
});