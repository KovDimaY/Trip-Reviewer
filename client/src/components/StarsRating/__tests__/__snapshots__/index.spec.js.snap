// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarsRating /> should render component 1`] = `
<div
  className="stars-rating-container"
>
  <label
    className="rating-label"
  >
    Rate your trip:
  </label>
  <Rating
    emptySymbol={
      <i
        className="far fa-star star-empty"
      />
    }
    fullSymbol={
      <i
        className="fas fa-star star-full"
      />
    }
    initialRating={undefined}
    onChange={[Function]}
    onHover={[Function]}
  />
</div>
`;
